src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1578,6 +1584,80 @@ private static bool HasAnonymousFunctionConversion(BoundExpression source, TypeS
 
             return IsAnonymousFunctionCompatibleWithType((UnboundLambda)source, destination) == LambdaConversionResult.Success;
         }
+
+        internal enum ConstructibleCollectionTypeKind
+        {
+            None = 0,
+            Array,
+            Span,
+            ReadOnlySpan,
+            CollectionInitializer,
+        }
+
+        internal static ConstructibleCollectionTypeKind GetConstructibleCollectionType(CSharpCompilation compilation, TypeSymbol destination, out TypeSymbol? elementType)
+        {
+            Debug.Assert(compilation is { });
+
+            if (destination is ArrayTypeSymbol arrayType)
+            {
+                if (arrayType.IsSZArray)
+                {
+                    elementType = arrayType.ElementType;
+                    return ConstructibleCollectionTypeKind.Array;
+                }
```

Comment:
prototype comment, what should we do here?  maybe a custom error about multidimensinoal arrays not being supported?

Reply:
I had a custom error message for multi-dimensional arrays but I removed it because the alternative error "Cannot initialize type 'object[*,*]' with a collection literal because the type is not constructible" seemed sufficient.